### PR TITLE
Add WebGL and CircleDrawer services

### PR DESCRIPTION
**Pull request by Nicholas-C-Brown**
Created an Abstract WebGL service for initializing the WebGL context, shaders, program and buffer.

Created a CircleDrawer service for drawing circles to the WebGL canvas.

### CODE REVIEW

File: src/app/services/WebGL/web-gl.service.ts [hunk 1]
```diff
@@ -0,0 +1,188 @@
...
+
+  private isInitialized(): boolean {
+    return this.gl != null
+      && this.vertexShader != null
+      && this.fragmentShader != null
+      && this.program != null
+      && this.buffer != null;
+  }
+
+  protected updateContext(){
```
Comment (EricShanks68) [inline src/app/services/WebGL/web-gl.service.ts:111]: Github is wanting a return type here. Is it right?

**Review by EricShanks68 — CHANGES_REQUESTED** [review]
Looks great Nic! Just wondering about the one thing github caught

File: src/app/services/WebGL/web-gl.service.ts (hunk 1 above)
Comment (EricShanks68) [reply src/app/services/WebGL/web-gl.service.ts:111]: Think it is implicitly turning to void, but should make it void explicit

Comment (Nicholas-C-Brown) [reply src/app/services/WebGL/web-gl.service.ts]: Yeah, the linter caught that. Should be fixed now.

Comment (EricShanks68) [reply src/app/services/WebGL/web-gl.service.ts]: blessssss

**Review by EricShanks68 — APPROVED** [review]

**Pull request merged**